projects/btldr/py_client/tests/test_btldr_hw.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+import pytest_check as check
+from unittest.mock import patch
+
+from can import Message as CANMessage
+from can.interface import Bus as CANBus
+
+from formula.projects.btldr.py_client.btldr import BtldrManager
+
+
+@pytest.fixture()
+def canbus():
+    canbus = CANBus(interface="socketcan", bitrate=500000, channel="can0")
+    yield canbus
+
+
+def test_ping_hw(canbus, ecu_id):
+    client = BtldrManager()
+    client.canbus = canbus
+
+    response = client.ping(ecu_id, 1)
+    check.is_not_none(response)
+
+
+def test_flash(canbus, ecu_id):
```

Comment:
Definitely not necessary for this PR, but would be nice to add a github action that runs any pytest/unit tests to ensure there's no regressions (I might look into this at some point).

Reply:
Yeah, if you wanted to look into this in the future, that'd be awesome! There are lots of opportunities for this.